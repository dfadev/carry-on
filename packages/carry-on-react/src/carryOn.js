/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import State from "./State";

export default (renderFn, stateProps) => {
  if (typeof renderFn === "object") {
    const actualStateProps = renderFn;
    renderFn = stateProps;
    stateProps = actualStateProps;
  }

  const CarryOnComponent = ({ from, debug, verbose, onMount, onUnmount, ...props }) => {
    const finalProps = { ...stateProps };
    if (from !== undefined) finalProps.from = from;
    if (debug !== undefined) finalProps.debug = debug;
    if (verbose !== undefined) finalProps.verbose = verbose;
    if (onMount !== undefined) finalProps.onMount = onMount;
    if (onUnmount !== undefined) finalProps.onUnmount = onUnmount;
    /* eslint-disable-next-line */
    if (props.id !== undefined) finalProps.id = props.id;

    return (
      <State {...finalProps}>
        {state =>
          renderFn.length === 1 ? renderFn(state) : renderFn(props, state)
        }
      </State>
    );
  };

  CarryOnComponent.displayName = (stateProps && stateProps.id) || "CarryOn";

  return CarryOnComponent;
};
