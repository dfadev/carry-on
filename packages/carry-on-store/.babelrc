{
  "presets": [
    ["env", {
      "modules": false,
      "targets": { "browsers": ["last 2 versions"] }
    }],
    "stage-2",
    "optimized"
  ],
  "env": {
    "development": {
      "plugins": ["external-helpers"],
    },
    "production": {
      "plugins": ["external-helpers"],
    },
    "test": {
      "presets": ["env", "stage-2"]
    }
  }
}
