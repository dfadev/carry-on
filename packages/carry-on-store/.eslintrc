{
  "parser": "babel-eslint",
  "extends": ["airbnb-base", "prettier"],
  "env": { "es6": true, "browser": true },
  "rules": {
    "no-empty-pattern": "off",
    "no-debugger": "warn",
    "no-unused-expressions": "off",
    "no-unused-vars": "warn",
    "no-plusplus": "off",
    "import/no-unresolved": "off",
    "no-restricted-syntax": "off",
    "no-underscore-dangle": "off",
    "prefer-template": "off",
    "spaced-comment": "off",
    "no-param-reassign": "off",
    "no-return-assign": "off",
    "no-multi-assign": "off",
    "one-var": "off",
    "prefer-destructuring": "off",
  }
}
