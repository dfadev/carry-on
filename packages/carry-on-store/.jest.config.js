module.exports = {
  //setupTestFrameworkScriptFile: "./node_modules/jest-enzyme/lib/index.js",
  setupTestFrameworkScriptFile: "./test/jest.setup.js",
  verbose: true,
  testPathIgnorePatterns: ["<rootDir>/build/", "<rootDir>/node_modules/", "<rootDir>/examples/", "<rootDir>/lib", "<rootDir>/src/jest.setup.js"],
  collectCoverage: true,
  coverageDirectory: "<rootDir>/coverage/",
  moduleNameMapper: {
    "^carry-on-utils$":
      "<rootDir>/node_modules/carry-on-utils/dist/carry-on-utils.esm.js",
  }
};
