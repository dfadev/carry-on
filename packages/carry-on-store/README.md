# carry-on-store

# STATUS: WIP SUBJECT TO CHANGE

- This is the core module that provides the store.

Read the documentation at: https://dfadev.github.io/carry-on
