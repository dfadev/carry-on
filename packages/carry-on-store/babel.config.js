module.exports = {
  presets: [
    [
      "@babel/preset-env",
      {
        modules: false,
        targets: {
          browsers: [">3%"]
        }
      }
    ]
  ],
  env: {
    test: {
      plugins: [
        "@babel/plugin-transform-modules-commonjs",
        [
          "@babel/plugin-transform-runtime",
          {
            regenerator: true
          }
        ]
      ],
      presets: ["@babel/preset-env"]
    }
  }
};
