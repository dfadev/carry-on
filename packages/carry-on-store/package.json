{
  "name": "carry-on-store",
  "version": "1.0.4",
  "description": "State manager",
  "source": "./src/index.js",
  "module": "./dist/carry-on-store.esm.js",
  "main": "./dist/carry-on-store.cjs.js",
  "browser": "./dist/carry-on-store.umd.js",
  "scripts": {
    "clean": "rimraf lib dist es coverage",
    "format": "prettier --write \"{src,test}/**/*.js\"",
    "format:check": "prettier --list-different \"{src,test}/**/*.js\"",
    "lint": "eslint src test",
    "pretest": "npm run build",
    "test": "cross-env BABEL_ENV=test jest --config ./.jest.config.js",
    "test:watch": "npm test -- --watchAll",
    "test:cov": "npm test -- --coverage",
    "build": "rollup -c rollup.config.js",
    "build-dev": "rollup -c rollup.config.js --configDebug",
    "watch": "npm run build -- -w"
  },
  "author": "Russ Panula <russ@dfa.net>",
  "license": "MIT",
  "devDependencies": {
    "@babel/core": "^7.12.10",
    "@babel/plugin-proposal-class-properties": "^7.12.1",
    "@babel/plugin-transform-modules-commonjs": "^7.12.1",
    "@babel/plugin-transform-runtime": "^7.12.10",
    "@babel/preset-env": "^7.12.11",
    "@testing-library/jest-dom": "^5.11.8",
    "babel-eslint": "^10.1.0",
    "babel-jest": "^26.6.3",
    "cross-env": "^7.0.3",
    "eslint": "^7.17.0",
    "eslint-config-airbnb-base": "^14.2.1",
    "eslint-config-prettier": "^7.1.0",
    "eslint-plugin-import": "^2.22.1",
    "jest": "^26.6.3",
    "prettier": "^2.2.1",
    "rimraf": "^3.0.2",
    "rollup": "^2.36.1",
    "rollup-plugin-babel": "^4.3.3",
    "rollup-plugin-eslint": "^7.0.0",
    "snapshot-diff": "^0.8.1"
  },
  "sideEffects": false,
  "gitHead": "e21942a1d6e23d964f9b74bd1c9c3744cb7b3c18",
  "dependencies": {
    "@babel/runtime": "^7.12.5",
    "carry-on-utils": "^1.0.4",
    "immer": "^8.0.0"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/dfadev/carry-on.git"
  }
}
