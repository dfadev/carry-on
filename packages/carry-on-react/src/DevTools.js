import { Component } from "react";
import { register, devTools } from "carry-on-store";

let hasRegistered = false;

export default class DevTools extends Component {
  constructor(props) {
    super(props);
    if (hasRegistered) return;
    hasRegistered = true;
    register(devTools(), props.store);
  }

  render() {
    return this.props.children || null;
  }
}
