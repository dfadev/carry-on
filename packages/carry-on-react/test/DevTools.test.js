/** @format **/
import React from "react";
import { initStores } from "carry-on-store";
import { render } from "react-testing-library";
import carryOn from "../src/carryOn";
import DevTools from "../src/DevTools";

afterEach(() => {
  initStores();
});

test("DevTools renders", () => {
  const App = () => (
    <DevTools>
      <div>ok</div>
    </DevTools>
  );

  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

test("Multiple DevTools renders", () => {
  const App = () => (
    <DevTools>
      <DevTools>
        <DevTools>
          <DevTools />
          <div>ok</div>
        </DevTools>
      </DevTools>
    </DevTools>
  );

  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
