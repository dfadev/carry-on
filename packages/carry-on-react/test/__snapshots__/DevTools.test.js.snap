// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevTools renders 1`] = `
<DocumentFragment>
  <div>
    ok
  </div>
</DocumentFragment>
`;

exports[`Multiple DevTools renders 1`] = `
<DocumentFragment>
  <div>
    ok
  </div>
</DocumentFragment>
`;
